tests/unit_tests/rigid_pendulum_poc/test_heavy_top.cpp

```diff
@@ -135,37 +135,45 @@ TEST(HeavyTopProblemFromBrulsAndCardona2010PaperTest, CalculateConstraintGradien
-         {0.775967, 0.63076, -0.00416521},
-         {-0.129909, 0.166271, 0.977485}}
-    );
-    auto angular_velocity_vector = create_vector({0.3, 0.1, 0.8});
-    auto position_vector = create_vector({0., 1., 0.});
-    auto lagrange_multipliers = create_vector({1., 2., 3.});
     const auto BETA_PRIME = 1.;
     const auto GAMMA_PRIME = 2.;
 
+    auto X0 = create_vector({0., 1., 0.});
+    auto rot0 = create_matrix({{1., 0., 0.}, {0., 1., 0.}, {0., 0., 1.}});
+    auto initial_position = multiply_matrix_with_vector(rot0, X0);
+    auto initial_orientation =
+        rotation_matrix_to_quaternion(RotationMatrix{{1., 0., 0.}, {0., 1., 0.}, {0., 0., 1.}});
+    auto gen_coords = create_vector({
+        initial_position(0),                       // component 1
+        initial_position(1),                       // component 2
+        initial_position(2),                       // component 3
+        initial_orientation.GetScalarComponent(),  // component 4
+        initial_orientation.GetXComponent(),       // component 5
+        initial_orientation.GetYComponent(),       // component 6
+        initial_orientation.GetZComponent()        // component 7
+    });
+
+    auto velocity = create_vector({0., 0., 0., 0.3, 0.1, 0.8});
+    auto lagrange_mults = create_vector({1., 2., 3.});
+    auto h = 0.1;
+    auto delta_gen_coords = create_vector({1., 1., 1.});
+
     auto iteration_matrix = heavy_top_iteration_matrix(
-        BETA_PRIME, GAMMA_PRIME, mass_matrix, inertia_matrix, rotation_matrix,
-        angular_velocity_vector, lagrange_multipliers, position_vector
+        BETA_PRIME, GAMMA_PRIME, gen_coords, velocity, lagrange_mults, h, delta_gen_coords
     );
 
     expect_kokkos_view_2D_equal(
         iteration_matrix,
         {
-            {15., 0., 0., 0., 0., 0., -1., 0., 0.},                                       // row 1
-            {0., 15., 0., 0., 0., 0., 0., -1., 0.},                                       // row 2
-            {0., 0., 15., 0., 0., 0., 0., 0., -1.},                                       // row 3
-            {0., 0., 0., -0.768003, 1.404458, -0.046875, 0.210962, -0.004165, 0.977485},  // row 4
-            {0., 0., 0., 0., 0.46875, 0., 0., 0., 0.},                                    // row 5
-            {0., 0., 0., 0.046875, 3.275712, -0.768003, -0.617251, -0.775967, 0.129909},  // row 6
-            {-1., 0., 0., 0.210962, 0., -0.617251, 0., 0., 0.},                           // row 7
-            {0., -1., 0., -0.004165, 0., -0.775967, 0., 0., 0.},                          // row 8
-            {0., 0., -1., 0.977485, 0., 0.129909, 0., 0., 0.}                             // row 9
+            {15., 0., 0., 0., 0., 0., -1., 0., 0.},                    // row 1
+            {0., 15., 0., 0., 0., 0., 0., -1., 0.},                    // row 2
+            {0., 0., 15., 0., 0., 0., 0., 0., -1.},                    // row 3
+            {0., 0., 0., -1.765625, 0.625000, -0.046875, 0., 0., 1.},  // row 4
```

Comment:
Why did these numbers change?

Reply:
I modified the function signature and had to redo the tests - values are verified against the prototype Fortran code after another round of revision.